packages/lib/modules/vebal/vote/VoteList/VoteListProvider.tsx

```diff
@@ -0,0 +1,89 @@
+'use client'
+
+import { createContext, PropsWithChildren, useMemo } from 'react'
+import { GetVeBalVotingListDocument } from '@repo/lib/shared/services/api/generated/graphql'
+import { useQuery } from '@apollo/client'
+import { useMandatoryContext } from '@repo/lib/shared/utils/contexts'
+import { usePoolListQueryState } from '@repo/lib/modules/pool/PoolList/usePoolListQueryState'
+import { useHiddenHandVotingIncentives } from '@repo/lib/modules/vebal/vote/hidden-hand/useHiddenHandVotingIncentives'
+import { useGaugeVotes } from '@repo/lib/modules/vebal/vote/gauge/useGaugeVotes'
+import { VotingPoolWithData } from '@repo/lib/modules/vebal/vote/vote.types'
+import { orderBy } from 'lodash'
+
+export function _useVoteList() {
+  // todo: implement vote's sorting/filtering
+  const queryState = usePoolListQueryState()
+
+  const {
+    data,
+    loading: voteListLoading,
+    previousData,
+    refetch,
+    error,
+  } = useQuery(GetVeBalVotingListDocument)
+
+  const voteListData =
+    voteListLoading && previousData
+      ? previousData.veBalGetVotingList
+      : data?.veBalGetVotingList || []
+
+  const pagination = queryState.pagination
+
+  const {
+    data: votingIncentives,
+    isLoading: votingIncentivesLoading,
+    error: votingIncentivesError,
+  } = useHiddenHandVotingIncentives()
```

Comment:
Any fetching that can be done server-side, e.g to external APIs or our own can be done in the RSC/layout component and passed into providers as inputs. That way the data is server side rendered and it speeds everything up.